spec/prometheus/client/counter_spec.rb

```diff
@@ -122,4 +117,57 @@
       end
     end
   end
+
+  describe '#with_labels' do
+    let(:expected_labels) { [:foo] }
+
+    it 'pre-sets labels for observations' do
+      expect { counter.increment }
+        .to raise_error(Prometheus::Client::LabelSetValidator::InvalidLabelSetError)
+      expect { counter.with_labels(foo: 'label').increment }.not_to raise_error
+    end
+
+    it 'registers `with_labels` observations in the original metric store' do
+      counter.increment(labels: { foo: 'value1'})
+      counter_with_labels = counter.with_labels({ foo: 'value2'})
+      counter_with_labels.increment(by: 2)
+
+      expect(counter_with_labels.values).to eql({foo: 'value1'} => 1.0, {foo: 'value2'} => 2.0)
+      expect(counter.values).to eql({foo: 'value1'} => 1.0, {foo: 'value2'} => 2.0)
+    end
+
+    context 'when using DirectFileStore' do
```

Comment:
How come only `Counter` has specific tests for `DirectFileStore`?

As an aside: this PR is even more encouragement for the test refactor I wanted to do. These tests definitely fall into the category that I'd prefer to test via a subclass of `Metric` rather than separately per type of metric.

Reply:
For the very reason you mentioned. I just need to test it in *one*, and `Counter` is kinda the default. We can definitely test it through a subclass of `Metric`.
I can also copy it to the other 3 metrics if you want, but it seemed a bit pointless

Reply:
Yeah, no point copying it multiple times. Let's have it in `Counter` for now, and then I'll go refactor the fuck out of these tests in a separate PR.

Reply:
Resolving this. I'll get it in that refactor of the metric tests.